packages/components/fields/checkbox-field/src/checkbox-field.tsx

```diff
@@ -0,0 +1,220 @@
+import {
+  createSequentialId,
+  getFieldId,
+  warning,
+} from '@commercetools-uikit/utils';
+import Stack from '@commercetools-uikit/spacings-stack';
+import FieldLabel from '@commercetools-uikit/field-label';
+import FieldErrors from '@commercetools-uikit/field-errors';
+import CheckboxInput from '@commercetools-uikit/checkbox-input';
+
+type TErrorRenderer = (key: string, error?: boolean) => ReactNode;
+type TFieldErrors = Record<string, boolean>;
+// Similar shape of `FormikErrors` but values are `TFieldErrors` objects.
+type TCustomFormErrors<Values> = {
+  [K in keyof Values]?: TFieldErrors;
+};
+
+const sequentialId = createSequentialId('checkbox-field-');
+
+const hasErrors = (errors?: TFieldErrors) =>
+  errors && Object.values(errors).some(Boolean);
+
+export type TCheckboxFieldProps = {
+  /**
+   * Used as HTML id attribute. An id is auto-generated when it is not specified.
+   */
+  id?: string;
+  /**
+   * A map of errors. Error messages for known errors are rendered automatically. Unknown errors will be forwarded to `renderError`.
+   */
+  errors?: TFieldErrors;
+  /**
+   * Called with custom errors.
+   * <br/>
+   * This function can return a message which will be wrapped in an ErrorMessage. It can also return null to show no error.
+   */
+  renderError?: TErrorRenderer;
+  /**
+   * Indicates if the value is required. Shows an the "required asterisk" if so.
+   */
+  isRequired?: boolean;
+  /**
+   * Indicates whether the field was touched. Errors will only be shown when the field was touched.
+   */
+  touched?: boolean;
+  /**
+   * Used as HTML name attribute of the input component
+   */
+  name?: string;
+  /**
+   * Value of the input component.
+   */
+  value?: string;
+  /**
+   * The checked property sets the checked state of the checkbox.
+   */
+  isChecked?: boolean;
+  /**
+   * If `true`, this state is shown as a dash in the checkbox, and indicates that its state is neither checked nor unchecked.
+   * This is most often used when the checkbox is tied to a collection of items in mixed states (E.g nested checkboxes).
+   * This takes precedence visually in case `isChecked` is marked as `true`
+   */
+  isIndeterminate?: boolean;
+  /**
+   *  Will be triggered whenever an `CheckboxInput` is clicked. Called with `event`
+   */
+  onChange: ChangeEventHandler<HTMLInputElement>;
+  /**
+   * Forces CheckboxInput to be rendered in a hovered state.
+   * Needed for cases when hovered appearance should be triggered by the parent component and not the CheckboxInput itself.
+   * CheckboxInput is capable of handling it's own hovering without the need to pass this prop.
+   */
+  isHovered?: boolean;
+  /**
+   * Disables the CheckboxInput
+   */
+  isDisabled?: boolean;
+  /**
+   * Makes the CheckboxInput readonly
+   */
+  isReadOnly?: boolean;
+  /**
+   * Indicates that the checkbox has an error
+   */
+  hasError?: boolean;
+  /**
+   * The descriptive text of the CheckboxInput, used as its label.
```

Comment:
I suppose we should be describing `CheckboxField` props (not `CheckboxInput`)

Reply:
Thank you for all the feedback. Updated here: https://github.com/commercetools/ui-kit/pull/2537/commits/0da451d5d93ec16a223a40ee9f6d27f940682a8b. 
There are still some design decisions follow-ups we are concluding on. So the PR is on wip for now until we conclude 🙌🏾